### PR TITLE
Fixes case insensitive complex env binding

### PR DESCRIPTION
**Pull request by jperkelens**
Hi! Sorry you've been getting so much activity from me, would have really liked to get this all in one PR. However, didn't think to look into this till after I discovered the insensitivity issue. This PR:

a) Allows for multiple env bindings deep in a complex binding where they share the same root. i.e.:
```python
v.bind_env('icings.regular', 'ICINGS_REGULAR')
v.bind_env('icings.premium', 'ICINGS_PREMIUM')
```

and

b) Respects the case insensitivity of env binding _while_ keeping key casing the same no matter what level of the config you pull.

### CODE REVIEW

File: vyper/vyper.py [hunk 1]
```diff
@@ -331,23 +347,27 @@ def _find(self, key):
...
-        env_key = self._env.get(key)
-
+        env_key = self._find_insensitive(key, self._env)
+        log.debug(f'Looking for {key} in env')
+        log.debug(self._env)
         if isinstance(env_key, list):
-            parent = next((self._config[real_key] for real_key in self._config.keys() if real_key.lower() == key), None)
+            parent = self._find_insensitive(key, self._config)
             found_in_env = False
+            log.debug(f'Found env key parent {key}: {parent}')
```
Comment (admiralobvious) [inline vyper/vyper.py]: Please change the log message to `'Found env key parent {0}: {1}'.format(key, parent)` so it works on older Python versions.

File: vyper/vyper.py [hunk 2]
```diff
@@ -331,23 +347,27 @@ def _find(self, key):
...
+
             for item in env_key:
-                print(item)
-                print(parent)
                 log.debug('{0} registered as env var parent {1}:'.format(key, item['env_key']))
                 val = self._get_env(item['env_key'])
                 
+
```
Comment (admiralobvious) [inline vyper/vyper.py]: Remove extra blank line

**Comment by admiralobvious** [comment]
No worries. Thanks for them! I made a change request with two very simple changes before I can merge it.

**Comment by jperkelens** [comment]
done!

**Pull request merged**